### DISCUSSION TITLE
Using Prisma Schema types as InputTypes or ArgsTypes?

### DISCUSSION THREAD

**Discussion by kleydon** [MichalLytek/typegraphql-prisma#348] [Q&A] [answered]
TypeGraphQL offers a way to "layer" `@InputType` on top of `@ObjectType`, to avoid duplication of similarly-shaped input/object types (at least for "simple" objects). 

Is there anything in this spirit, that enables use of prisma schema types as (or within) `@InputType`s or `@ArgsType`s?

**Comment by MichalLytek** [comment, marked as answer]
No, generated resolvers, args and inputs reflects the rich Prisma Client API. Having simple input like outputs means you just don't need the generator as you just have simple actions.

**Reply by kleydon** [reply]
Just realized that it _is_ possible to use the generated `<model-name>CreateInput` as an input (overlooked the obvious) and, alternatively, maybe it is possible to apply `@InputType('<model-name>Input')` via `applyModelsEnhanceMap()` as well...

This library is such a great resource. Thanks for putting it out there!
